### PR TITLE
Throw if a watchdog is closed before it is stopped.

### LINKED ISSUE #7: Make WatchDog.close complain if not stopped
If calling close on a WatchDog the application programmer could thing that the dog has also stopped.

Can we make close throw if the dog is not stopped, unless passed with a --force flag?

### PR DESCRIPTION
**Pull request by floitsch**
Fixes #7.

### CODE REVIEW

File: tests/throw-close-test.toit [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import expect show *
+import watchdog show WatchdogServiceClient
+
+import .util
+
+main:
+  run-test: |client ms system-dog| test-recover client ms system-dog
+
+// If a watchdog gets lost/closed, another watchdog with the same name can take over.
+test-recover client/WatchdogServiceClient ms/int system-dog/FakeSystemWatchdog:
```
Comment (kasperl) [inline tests/throw-close-test.toit:14]: What is this recovering? I don't understand the method name or the comment?

File: src/watchdog.toit [hunk 2]
```diff
@@ -140,5 +142,14 @@ class Watchdog extends ServiceResourceProxy:
   Does nothing if the watchdog is not started.
   */
   stop -> none:
+    is-stopped_ = true
     client := (client_ as WatchdogServiceClient)
     client.stop_ handle_
+
+  /**
+  See $super
```
Comment (kasperl) [inline src/watchdog.toit:150]:
```suggestion
  See $super.
```

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: tests/throw-close-test.toit (hunk 1 above)
Comment (floitsch) [reply tests/throw-close-test.toit]: bad copy/paste from existing test...
done.

File: src/watchdog.toit (hunk 2 above)
Comment (floitsch) [reply src/watchdog.toit]: done.

**Pull request merged**